src/errorcorrection.jl

```diff
@@ -100,49 +145,55 @@ function *(a::Poly, b::Poly)::Poly
 end
 
 """
-    ÷(f::Poly, g::Poly)
-
-Quotient of Euclidean division.
-"""
-function ÷(f::Poly, g::Poly)
-    quodeg = length(f) - length(g) ## degree of the quotient polynomial
-    quodeg < 0 && return Poly([0])
-    g <<= quodeg
-    gn = lead(g) ## leading term of g(x)
-    quocoef = Vector{Int}(undef, quodeg + 1)
-    for i in 1:quodeg
+    euclidean_divide(f::Poly, g::Poly)
+
+Returns the quotient and the remainder of Euclidean division.
+"""
+function euclidean_divide(f::Poly, g::Poly)
+    ## remove trailing zeros
+    g, f = rstripzeros(g), rstripzeros(f)
+    g == Poly([0]) && throw(DivideError())
```

Comment:
Generally, we don't use the raw number `1`/`0` to allocate the array because it will be `Vector{Int}` and would be troublesome for cases that Float64 is used. Instead it is usually something like `fill(zero(eltype(f.coeff), 1)`.

But here I assume it's always `Vector{Int}` here? If so then the plain `[0]` is absolutely okay.

Reply:
In general, the well defined coefficiences involved in Poly range from 0 to 255.

```jl
struct Poly
    coeff::Vector{Int}
end
```

It may be better to shift from type Int to UInt8. (need some checks)

Reply:
For a highly-optimized algorithm where bandwidth is the bottleneck, `UInt8` vs `Int` can make a lot of difference in performance, e.g., [dilation](https://github.com/JuliaImages/ImageMorphology.jl/pull/102#issuecomment-1165297247). But for QRCoders, it might be too early for this to make a difference.